standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ExceptionHandler.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore;
+
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
+import org.apache.hadoop.hive.metastore.utils.JavaUtils;
+import org.apache.thrift.TException;
+
+import static java.util.Objects.requireNonNull;
+
+public final class ExceptionHandler {
+  private final Exception e;
+
+  private ExceptionHandler(Exception e) {
+    this.e = e;
+  }
+
+  public static ExceptionHandler handleException(Exception e) {
+    requireNonNull(e, "Exception e is null");
+    return new ExceptionHandler(e);
+  }
+
+  /**
+   * Throws if the input e is the instance of the class clz
+   */
+  public <T extends Exception> ExceptionHandler
+      throwIfInstance(Class<T> t) throws T {
+    if (t.isInstance(e)) {
+      throw t.cast(e);
+    }
+    return this;
+  }
+
+  /**
+   * Throws if the input e is the instance of the class clzt or  class clze in order
+   */
+  public <T extends Exception, E extends Exception> ExceptionHandler
+      throwIfInstance(Class<T> t, Class<E> e) throws T, E {
+    throwIfInstance(t);
+    throwIfInstance(e);
+    return this;
+  }
+
+  /**
+   * Throws if the input e is the instance of the class clzt or  clze or clzc in order
+   */
+  public <T extends Exception, E extends Exception, C extends Exception> ExceptionHandler
+      throwIfInstance(Class<T> t, Class<E> e, Class<C> c) throws T, E, C {
```

Comment:
I see that you have a 2 argument and 3 argument version of this. Could we improve this by
public <T extends Exception> ExceptionHandler throwIfInstance(Class<T> ... te) throws T {
   for (Class<T> t : te) {
       throwIfInstance(t);
    }
    return this;
}

Reply:
Yes, we can do it by this more simplified way, thank you very much for the comments!